### ISSUE TITLE
ClientOnly: using async methods breaks defineExpose

### ISSUE THREAD

**Issue by vphilot** [nuxt/nuxt#15086]
### Environment

- Operating System: `Darwin`
- Node Version:     `v16.13.0`
- Nuxt Version:     `3.0.0-rc.11`
- Nitro Version:    `0.5.4`
- Package Manager:  `npm@8.1.0`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://codesandbox.io/p/github/nuxt/starter/csb-zd6kiy/draft/nostalgic-murdock

### Describe the bug

Implementing any async operations in a Client Only component will break `defineExpose` and the component instance won't have its exposed method available to parent anymore. 

E.g.

ChildComponent.vue
```vue
<script setup>
  const { data, pending, error, refresh } = await useFetch(
    () => "https://api.nuxtjs.dev/mountains"
  )

  const childMethod = () => {
     // do something
  }

  defineExpose({ childMethod })
</script>
```

ParentComponent.vue
```vue
<template>
  <ClientOnly>
    <ChildComponent ref="childRef" />
  </ClientOnly>
</template>

<script setup>
  const childRef = ref()

  // ...
  childRef.value.childMethod()
  // ERROR: childRef.value.childMethod is not a function
</script>
```

### Additional context
If you remove the `useFetch` call on `childComponent`, the exposed child method is available in the parent again

**Comment by Ellipse120** [comment]
Any updates for this issue ?

**Comment by danielroe** [comment]
I would follow the upstream issue for updates instead.

**Comment by ilyaDegtyarenko** [comment]
Some related info:

<img width="756" alt="image" src="https://github.com/nuxt/nuxt/assets/37239202/3fffef59-900b-4a5a-b8be-233d259ccc0a">

**Comment by bayramorhan** [comment]
**Works**
```
const onDataRefreshed = async () => {
	await itemListRef.$.exposed.refreshData()
}
```

**Not Works**
```
const onDataRefreshed = async () => {
	await itemListRef.value.refreshData()
}
```

**Comment by dargmuesli** [comment]
fyi, upstream was fixed in v`3.5.0-alpha.4` :rocket:

**Comment by wister** [comment]
Nuxt: `3.13.2`
It may have regressed, here's a reproduction: https://stackblitz.com/edit/github-ffcqc6
Thanks
